platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/attribution/AttributionMeasure.java

```diff
@@ -0,0 +1,113 @@
+package com.mapbox.mapboxsdk.attribution;
+
+import android.graphics.Bitmap;
+import android.graphics.PointF;
+import android.widget.TextView;
+
+public class AttributionMeasure {
+
+  private Bitmap logo;
+  private Bitmap logoSmall;
+  private Bitmap snapshot;
+  private TextView textView;
+  private TextView textViewShort;
+  private float margin;
+
+  private boolean shorterText;
+
+  AttributionMeasure(Bitmap snapshot, Bitmap logo, Bitmap logoSmall, TextView tv, TextView tvShort, float margin) {
+    this.snapshot = snapshot;
+    this.logo = logo;
+    this.logoSmall = logoSmall;
+    this.textView = tv;
+    this.textViewShort = tvShort;
+    this.margin = margin;
+  }
+
+  public AttributionLayout measure() {
+    float logoContainerWidth = logo.getWidth() + (2 * margin);
```

Comment:
`measure()` is kinda long. What about extracting each block of code into a private method?